eventlet/hubs/asyncio.py

```diff
@@ -144,3 +165,37 @@ async def async_run():
                 self.stopping = False
 
         self.loop.run_until_complete(async_run())
+
+    async def wait(self, seconds=None):
```

Comment:
You're duplicating a whole bunch of `asyncio` logic, and moreover doing it in a way that makes it much less scalable because e.g. you're not using `epoll` on Linux. And the issues with `select()`'s max file descriptor limit. This does not seem like the way to solve this problem.

Reply:
Or, let me rephrase. You are completely breaking the `asyncio` event loop by blocking. So now any normal `asyncio` code will only be run intermittently.

Reply:
That could explain the failure I observed on the asyncio tests you originally implemented. See the job failure.